tests/lean/defInst.lean.expected.out

```diff
@@ -1,4 +1,4 @@
-[4, 5, 6]
+<missing Repr instance for Foo>
```

Comment:
Regression? Maybe okay since `Foo` is a non-reducible type alias?

Reply:
I guess you could call it a regression.  We used to reduce the types if we couldn't find an `Eval` instance.  Now there's always an `Eval` instance so we can't do that anymore (easily).

Personally I think it's the right tradeoff.  Getting type class instances for the reduced type is completely at odds with every other part of Lean.  You're supposed to be able to use `def` to create type wrappers with different instances, like a `def WithZero (A : Type u) := Option A` where `none` should always be printed as `0`.